### DISCUSSION TITLE
Using a GND pin group with more than one signal group creates an error

### DISCUSSION THREAD

**Discussion by ghost** [ansys/pyedb#961] [Q&A] [answered]
I am trying to create a port from two signals to a common GND pin group, However, I get an error message:

> ValueError: Failed to create terminal.

A minimially reproduced example is derived from a [User guide](https://edb.docs.pyansys.com/version/stable/user_guide/excitations/create_circuit_ports_on_component.html) example:

```
from pyedb import Edb
from pyedb.generic.general_methods import generate_unique_folder_name
import pyedb.misc.downloads as downloads

temp_folder = generate_unique_folder_name()
targetfile = downloads.download_file("edb/ANSYS-HSD_V1.aedb", destination=temp_folder)
edbapp = Edb(edbpath=targetfile, edbversion="2024.2")
print(targetfile)

# create pin groups
edbapp.siwave.create_pin_group_on_net("U1", "1V0", "PG_V1P0_S0")
edbapp.siwave.create_pin_group_on_net("U1", "GND", "PG_GND")
# create port on pin group
edbapp.siwave.create_circuit_port_on_pin_group("PG_V1P0_S0", "PG_GND", impedance=50, name="test_port_1")

#create second signal pingroup & port on net to GND group

edbapp.siwave.create_pin_group_on_net("U1", "AVCC_1V3", "PG_AVCC_1V3")
edbapp.siwave.create_circuit_port_on_pin_group("PG_AVCC_1V3", "PG_GND", impedance=50, name="test_port_2")

edbapp.save_edb()
edbapp.close_edb()
```

At the circuit port created on PG_AVCC_1V3, the failure is seen; if either create_circuit_port_on_pin_group line is commented out, the code completes normally. Alternately, I could add a separate GND pin group for each signal pin group, however this will raise a warning during the solve about pins in multiple pin groups.

Is there a different way to connect two signals to a common GND group?

**Comment by kestne** [comment]
Hi, I am getting a similar error. I just posted a question on this error as well.

**Reply by svandenb-dev** [reply, marked as answer]
Hi @kestne,
I think you got also this error because the terminal for the reference is already defined with same name. I would create another pin group, you will get a warning but since you are sure you are not shorting different nets I believe you can ignore.
Alternatively you can create a single pingroup but create port on single pins instead on pingroup. EDB support pingroup terminal and padstack instance terminal. 
here is a code snipet which could be inspiring:
```
reference_pin = pin.get_reference_pins(reference_net="GND",
                                                   search_radius=5e-3,
                                                   max_limit=1,
                                                   component_only=True)
positive_terminal = pin.get_terminal(create_new_terminal=True)
positive_terminal.is_circuit_port = True
if not reference_pin[0].terminal:
    negative_terminal = reference_pin[0].get_terminal(create_new_terminal=True)
else:
    negative_terminal = reference_pin[0].terminal
positive_terminal.ref_terminal = negative_terminal
```

**Reply by ghost** [reply]
I apologize, I did not recognize your entire comment was not directed towards kestne. My initial concern with this implementation was an edge case that shouldn't really matter, or it would be a problem either way... I will distil inspiration best I can.

**Reply by ghost** [reply]
actually, the second way is better, right? I can just create the pin groups, then add ports as pins and it will be connected to the pin and therefore the pin group? In this way I won't have to create a length-one pingroup for a positive pin, in, say, a SI sim

**Comment by ghost** [comment]
Here's how I ended up rewriting this:


```
from pyedb import Edb
from pyedb.generic.general_methods import generate_unique_folder_name
import pyedb.misc.downloads as downloads

temp_folder = generate_unique_folder_name()
targetfile = downloads.download_file("edb/ANSYS-HSD_V1.aedb", destination=temp_folder)
edbapp = Edb(edbpath=targetfile, edbversion="2024.2")
print(targetfile)

nets = ['1V0', 'AVCC_1V3']
edbapp.siwave.create_pin_group_on_net("U1", "GND", "PG_GND")
for net in nets:
    edbapp.siwave.create_pin_group_on_net("U1", net, f"PG_{net}")
    pos_pin_dict = edbapp.components.get_pins('U1', net)
    pos_pin = pos_pin_dict[list(pos_pin_dict.keys())[0]]
    ref = pos_pin.get_reference_pins(reference_net = 'GND', component_only = True)
    pos_term = pos_pin.get_terminal(create_new_terminal=True)
    if not ref[0].terminal:
        negative_terminal = ref[0].get_terminal(create_new_terminal=True)
    else:
        negative_terminal = ref[0].terminal
    pos_term.ref_terminal = negative_terminal
    pos_term.name = f'{net}_PG_U1'

edbapp.save_edb()
edbapp.close_edb()
```

Thank you for the inspo
